# DREIMT Help

Drug REpositioning for IMmune Transcritome
## Index
1. [What is DREIMT?](help#whats-dreimt)
    1. [How it works?](help#how)
    2. [Dreimt tools](help#tools)
2. [How to select drug candidates?](help#how-candidate)
    1. [Drug prioritization score (Tau)](help#tau)
    2. [False discovery rate](help#fdr)
    3. [Drug specificity score](help#dss)
3. [Database](help#database)
    1. [Database fields](help#fields)
    2. [Database filtering](help#filtering)
    3. [DREIMTdb query example I](help#query-db1)
    4. [DREIMTdb query example II](help#query-db2)
4. [Drug prioritization](help#Drug-prioritization)
    1. [Input](help#queries-input-prioritization)
    2. [Drug prioritization example](help#example1)
5. [Signature comparison](help#comparison)
    1. [Input](help#queries-input-comparison)
    2. [Signature comparison example I](help#comparison-example-1)
    3. [Signature comparison example II](help#comparison-example-2)
6. [Input file](help#inputs)
7. [Queries history](help#queries-history)
8. [Error messages](help#errors)
9. [FAQS](help#faqs)
    1. [Drugs in DREIMT](help#f1)
    2. [Tau score change](help#f2)
    3. [Signature out of the required range](help#f3)
    4. [Tau interpretation](help#f4)
    5. [DREIMT terminology](help#f5)



## 1. What is DREIMT?<a name="whats-dreimt"></a>

DREIMT is a bioinformatics tool for **hypothesis generation** and prioritization of **drugs capable of modulating immune cells activity**. DREIMT aims to repurpose drugs capable of modulating immune cells from transcriptomics data.

DREIMT integrates **4,690 drug profiles** from The Library of Network-Based Cellular Signatures (LINCS) L1000 data set  and **2,700 manually curated immune gene expression signatures** from multiple sources to generate a drug-immune signature association database.


#### 1.1 How it works?<a name="how"></a>
DREIMT uses the rationale developed by the LINCS L1000 project (Subramanian et al. 2017) based in the use of geneset enrichment analysis (GSEA) methodologies to compare gene expression signatures to a set of drug transcriptomics profiles. The GSEA score (ES) is then prioritized against a large set of ES to generate an association score (tau) for each drug profile-immune signature pair.

This methodology is employed to create a prioritized drug association database for immune gene expression signatures and to prioritize drugs from user-provided immune expression signatures. Complete description of the methodology can be found the in supplementary material.

<img src="assets/help/assets/help/SplashImage.png" alt="Abstract figure"></img>

#### 1.2 DREIMT tools<a name="tools"></a>
DREIMT contains three tools accessible in the upper menu.

- **Drug prioritization**: to retrieve ranked drug associations for user-provided immune gene expression signatures.
- **Signature comparison**: to compare user-provided immune gene expression signature to DREIMT database immune signatures.
- **Database**: precalculated ranked drug associations for a set of 2,700 manually curated immune gene expression signatures retrieved from the literature.

<img src="assets/help/assets/help/Splash_ppt.png" alt="Splash"></img>

DREIMT also contains three sections of interest.
- **Help**: information and examples for the correct use of DREIMT.
- **REST API**: information for the use of the REST API tool.
- **Queries history**: stored results for drug prioritization and signature comparison tools.













## 2. How to select drug candidates?<a name="how-candidate"></a>
DREIMT offers three scores to rank and prioritize drug candidates.

#### 2.1 Drug prioritization score (Tau)<a name="tau"></a>
The Drug prioritization score (**tau**) indicates the degree of specificity for a given drug profile-immune signature association pair compared to a large set of drug-immune signatures associations pairs in DREIMT's database.

Tau score ranges from **-100 to 100**, extreme values, both positive and negative, are scores of high specificity. The tau score measures:
- How high is a particular drug profile-immune gene expression association compared to the rest of association for a given drug.
- How high is a particular drug profile-immune gene expression association compared to all the associations calculated for a given immune signature.

For an **initial search** of drug candidates **tau scores >|90|** are considered as strong hypotheses for the drug profile-immune signature association. Tau score should be considered as the **primary score** to prioritize drug associations.

<img src="assets/help/assets/help/Sign_help.png" alt="Tau sign"></img>

#### 2.2 False discovery rate<a name="fdr"></a>
**FDR** (Benjamini Y. and Hochberg Y., 1995) are generated by GSEA to account for multiple hypothesis testing. Drug profile-immune gene expression signature associations with FDR < 0.05 are considered statistically significant.

FDR values are not intended to be used as cut-off values, but to prioritize associations when multiple drugs are associated with a signature. FDR values can also be helpful to retrieve meaningful information from the drug profile-immune gene expression signature associations. When using a complete signature (composed by both the upregulated and downregulated genes), FDR values can indicate which of the two genesets accounts for the highest weight in the drug profile-immune gene expression signature association, thus helping to retrieve biological information.

#### 2.3 Drug specificity score<a name="dss"></a>
The Drug specificity score (**DSS**) adapted from Hodos et al. (Hodos et al. 2018) summarizes the replicability of a given drug profile across multiple cancer cell lines. This score considers the case whether the response to a drug is cell-specific or comparable across cells.

DSS scores ranges from **0 to 1**, scores close to 1 indicate that the transcriptomic effect of a given drug is found to be akin all tested cancer cell lines in LINCS database, on the contrary, scores close to 0 indicate the drug effect has been observed to be more cancer cell line specific.

If multiple drugs are associated (tau score >|90|) with an immune expression signature, DSS can be used to prioritize drugs. Not all drugs have DSS available.

















## 3. Database<a name="database"></a>
DREIMT database (DREIMTdb) contains **precalculated drug associations** for nearly **2,700 manually curated immune gene expression signatures** retrieved from the literature and **4,690 drug profiles** from The Library of Network-Based Cellular Signatures (LINCS) L1000 data set.

DREIMTdb in its **online** version only offers drug profile-immune gene expression signature associations with **tau > |80|**.

#### 3.1 Database fields<a name="fields"></a>
DREIMTdb contains 12 information fields:

- **Drug**: drug common name and link to PubChem. Hovering the mouse over will display a tooltip with the drug BROAD ID and genes target by the drug.
- **Summary**: summarized sentence that includes relevant manually annotated information to ease with drug association interpretation.  
- **Case type**: cell type/cell subtype used as case for gene expression signature generation. If cell type and cell subtype names match, only cell subtype name is shown.
- **Reference type**: cell type/cell subtype used as reference for gene expression signature generation. If cell type and cell subtype names match, only cell subtype name is shown.
- **Signature info**: immune signature source name and a pop-up panel with signature related links to: signature source database, DREIMTdb summarized results and NCBI data generation publication.
- **Up genes FDR**: adjusted p-value for the upregulated geneset.
- **Down genes FDR**: adjusted p-value for the downregulated geneset.
- **Tau**: drug prioritization score, range (-100, 100).
- **DSS**: drug specificity score, range (0, 1).
- **Drug status**: current known status of drug (approved, experimental or withdrawn).
- **Drug MOA**: drug mechanism of action.
- **More info.**: symbolic signature information of: organism source of the signature (*Homo sapiens*, *Mus musculus*), experimental design (*in vitro, in vivo, ex vivo, patient, in silico*) and signature treatment (signature, upregulated signature, downregulated signature and geneset).


<img src="assets/help/assets/help/DatabaseSummary_ppt.png" alt="Database summary"></img>

Depicted: Drug fields (orange), immune signature fields (green), drug-signature association fields (blue) and fields with tooltips (red asterisk).


#### 3.2 Database filtering<a name="filtering"></a>

DREIMTdb is queried using two types of filters; **basic** and the **advanced** filters.

Clicking over each filter box displays a panel with all available options, boxes allow text input and will display options matching the input. Once a filter is applied the rest of filters will automatically update to only show feasible filtering options.

DREIMTdb results can be downloaded in csv format for queries with less than 1,000 results. DREIMTdb can be queried programatically using the API.

##### 3.2.1 Basic filters
**By default DREIMTdb prompts the basic filters panel:**

- **Cell type/subtype 1**: filters drug associations by the specified immune cell type/cell subtype of interest.
- **Drug effect**: specifies the modulatory effect of interest (boosting/inhibition) on the selected cell type in "Cell type/subtype 1" filter. This filter is activated after a cell type has been specified in "Cell type/subtype 1".
- **Drug name**: filters drugs associations by the specified drug name. This filter can be used both,  independently and jointly with the other two basic filters.

<img src="assets/help/assets/help/Basic_filters_ppt.png" alt="Basic filters"></img>

Both, basic and advanced **filters** can be cleared by clicking the **Clear filters** button.


##### 3.2.1 Advanced filters
Clicking in the Advanced filters displays a panel of detailed filters that can be used to refine the query. Previously selected basic filters will be displayed in the advanced filters panel, unless have been previously erased.

**Advanced filters are organized in three main categories:**

##### Score filters
- **Min. tau (1)**: filters drug associations by the absolute tau score selected.
- **Max. up genes FDR (2)**: sets a threshold FDR value for the upregulated geneset of the immune signature.
- **Max. down genes FDR (3)**: sets a threshold FDR value for the downregulated geneset of the immune signature.
- **Interaction type (4)**: sets the use given to the immune gene expression signature to calculate the drug association score: signature (uses the complete signature composed of the upregulated and downregulated genes), signature up (uses only the upregulated genes), signature down (uses only the downregulated genes), geneset (genes without specified direction).

<img src="assets/help/assets/help/Adv_filters_row1.png" alt="Advanced R1"></img>

##### Drug filters
- **Drug effect (1)**: (basic filter) specifies the modulatory effect of interest (boosting/inhibition) on the selected cell type in the Cell type/subtype 1 filter. This filter is activated after a cell type has been specified in Cell type/subtype 1 filter.
- **Drug name (2)**: (basic filter) filters drugs associations by the specified drug name.
- **MOA (3)**: filters associations by the specified mechanism of action of the drugs.
- **Drug status (4)**: filters by the specified status of the drug.
- **Min. drug specificity score (5)**: specifies a minimum threshold value for the DSS score.

<img src="assets/help/assets/help/Adv_filters_row2.png" alt="Advanced R2"></img>

##### Signature/Geneset filters
- **Name (1)**: filters associations by the immune signature source name.
- **Source DB (2)**: filters associations by the immune signatures source database.
- **PubMed ID (3)**: filters associations by the PubMed ID of the publication that generated the immune signature data.
- **Experimental design (4)**: filters associations by the experimental design performed to generate the immune signatures (in vivo, in vitro, transfection, ex vivo, patient or in silico).
- **Cell type/subtype 1 (5)**: (basic filter) filters associations in which the specified immune cell type/cell subtype is involved.
- **Cell type 1 disease (6)**: filters by disease source of the "Cell type/subtype 1". This filter is activated after a cell type has been specified in Cell type/subtype 1 filter.
- **Cell type 1 treatment (7)**: filters by perturbations applied to the selected "Cell type/subtype 1". This filter is activated after a cell type has been specified in Cell type/subtype 1 filter.
- **Cell type/subtype 2 (8)**: allows to specify a second immune cell type involved in the generation of the immune signature. This filter is activated after a cell type has been specified in Cell type/subtype 1 filter. It will filter signatures that jointly contains the cells specified as subtype 1 and 2.
- **Condition (9)**: filters signatures in which the specified disease/treatment is involved.
- **Organism (10)**: selects organism source of the signature: *Homo sapiens*, *Mus musculus*.

<img src="assets/help/assets/help/Adv_filters_row3.png" alt="Advanced R3"></img>

**Important**: Notice that "Cell type/subtype 1" and "Cell type/subtype 2" only indicates cells involved in a signature, it does not specify which cell type is used as case and which as reference (see [example II for a practical demonstration](help#query-db2))

#### 3.3 DREIMTdb query example I<a name="query-db1"></a>
In this example we will perform a query to search for drugs with the potential to inhibit the effect of **TOX overexpression** (exhaustion marker) **in T CD8+ cells**. For this example we are only interested in associations made with the complete immune signature (upregulated and downregulated genes).


##### Filters applied
- **Cell type/subtype 1**: T cell/ T CD8+.
- **Cell type 1 treatment**: Overexpression [TOX].
- **Drug effect**: Inhibition.
- **Interaction type**: Signature.
- **Min. tau:** 90.

<img src="assets/help/assets/help/Example_1_signature_ppt.png" alt="Example 1"></img>

This query yields a single result, **BIX-01294** (an experimental methyltransferase inhibitor), with a tau score of -91.08. Both genesets are statistically significant (upregulated: 0.014 FDR, downregulated: 0.043 FDR), which indicates that both genesets have been highly relevant for the association with the drug.

The icons in the More information field indicate *Mus musculs* as the source organism for the immune cells, the experiment was performed *in vitro* and the complete signature (up and downregulated genes) is being accounted for the drug association.

The summary sentence indicates that this drugs might **inhibit** those **T CD8+ cells simulated with LCMV and with TOX overexpression compared to those T CD8+ cells stimulated with LCMV**. This drug targets EHMT2 and EHMT1, a search in the literature indicates that these genes are known to be related with CD8+ exhaustion (Birili et al., 2020).

Hovering the mouse over the Signature information field will display a tooltip with detailed information such the PubMed ID and title of publication source of the signature.

<img src="assets/help/assets/help/TooltipExample1.png" alt="Tooltip ex1"></img>

Clicking over the link in the Signature information field displays a panel with three useful links.  

- **PubMed**: a link to the publication that generated the expression data to generate the immune signature.
- **DREIMT Query**: summarized results for the selected immune signature.
- **Signature source**: a link to the source of the immune signature, in cases where the signature has been obtained from the Literature and not from a database this will link to the original publication.

<img src="assets/help/assets/help/Example_1_siganture_detailed_1.png" alt="Example1 detailed"></img>

By selecting the **DREIMT Query** option a new page with detailed information for the current signature will be displayed.


**This page displays three summary plots and a table.**
- **Drug prioritization plot (1)**: plot representing all drugs associated (|tau| > 80) with the immune signature. The x-axis represents the tau score, y-axis the -log10(FDR).

 Two areas of interest are depicted in the plot to prioritize drug selection:
 - **Best candidates**: drug associations with |tau| > 90 and a statistically significant FDR (FDR < 0.05).
 - **Good candidates**: drug associations with |tau| > 90 but no statistically significant FDR (FDR > 0.05).

 By default all associations are represented:
 - **Circles**: drug associations using the complete signature (up and down genes)
 - **Upright triangles**: drug associations performed using only the upregulated signature.
 - **Upside triangles**: drug associations performed using only the downregulated signature.
 - **Squares**: drug associations performed using genesets.

- **Drugs by approval status (2)**: pie-chart representing the status of drugs associated with the immune signature.
- **Drugs by MOA (3)**: pie-chart representing the mode of action of drugs associated with the immune signature.
- **Drug association table (4)**: summarized database with associations for the specified immune signature. The header contains filter boxes, once applied, the table will filter associations and all plots will update to show filtered results.

Plots can be downloaded in multiple formats (png, jpeg, pdf and svg) using the download buttons. Hovering the mouse over the the plots will display tooltips with detailed information.

<img src="assets/help/assets/help/PlotsSummarized_filter.png" alt="Plot summary"></img>

After applying filters in the header of the to keep only associations made with the complete signature, |tau| > 90 and experimental drugs the plots automatically update to show table filtered associations. Hovering the mouse over each point in the plot will activate a tooltip.

<img src="assets/help/assets/help/PlotsSummarized.png" alt="Plot summary2"></img>



#### 3.4 DREIMTdb query example II<a name="query-db2"></a>
In this example we will perform a query to search for approved drugs with the potential to boost human macrophage cell polarization towards a macrophage M1 phenotype instead of to macrophage M2 phenotype. Firstly we will need to filter immune signatures generated from the comparison of macrophages M1 and M2, then apply the rest of filters to narrow our search.

##### Filters applied (option I)
- **Cell type/subtype 1**: Macrophage/Macrophage M1.
- **Cell type/subtype 2**: Macrophage/Macrophage M2.
- **Organism**: *Homo sapiens*.
- **Drug effect**: Boosting.
- **Drug status**: Approved.

For this filtering we have selected "Boosting" as drug effect, because the "Drug effect" filter works on the cell type specified in the "Cell type/subtype 1" filter, which, in this case is Macrophage M1.

<img src="assets/help/assets/help/MacrophagesDB1_ppt.png" alt="Macrophages DB1"></img>

##### Filters applied (option II)
- **Cell type/subtype 1**: Macrophage/Macrophage M2.
- **Cell type/subtype 2**: Macrophage/Macrophage M1.
- **Organism**: *Homo sapiens*.
- **Drug effect**: Inhibition.
- **Drug status**: Approved.

For this filtering we have selected "Inhibition" as drug effect, because the "Drug effect" filter works on the cell type specified in the "Cell type/subtype 1" filter, which, in this case is Macrophage M2.

<img src="assets/help/assets/help/MacrophagesDB2_ppt.png" alt="Macrophages DB2"></img>


Both queries yield the same results, seven drugs that potentially could inhibit the M2 phenotype/boost M1 phenotype polarization. In this case filtered associations have been made using the down regulated geneses from the classical M1 macrophages vs alternative M2 macrophages signature.

<img src="assets/help/assets/help/MacrophageQueryResults.png" alt="Macrophages results"></img>













## 4.  Drug prioritization<a name="Drug-prioritization"></a>
Drug prioritization tool performs drug association prioritizations from user-provided immune gene expression signatures against 4,690 DREIMTdb drug profiles.

#### 4.1 Input<a name="queries-input-prioritization"></a>

- **Query title**: (optional) name given to the query. This name will be used to store the results in the Queries history panel.
- **Case cell description**: (optional) brief description of the case immune cell used to generate the signature. It is highly recommended to add a short description, this will be used to create a brief sentence to facilitate the interpretation of the results.
- **Reference cell description**: (optional) brief description of the reference immune cell used to generate the signature. It is highly recommended to add a short description, this will be used to create a brief sentence to facilitate the interpretation of the results.
- **Signature descriptor**: this tab indicates the type of input being queried: signature, signature up, signature down or geneset. If genes are input directly to the text boxes, the tab must be manually specified. Once a descriptor is specified, the tool automatically blocks the input of data in certain boxes to avoid incorrect inputs (e.g.. writing a downregulated signature in the upregulated text box). If the input is uploaded as csv/tsv file, the descriptor tab will automatically select the correct option based in the csv file content.
- **Query genes**: user-provided immune signature, can be provided as a list of genes pasted to the text boxes or as a csv/tsv file. For a complete description of input formats see [Inputs file](help#inputs).

<img src="assets/help/assets/help/DrugPrioritization_panel1.png" alt="Drug prioritization"></img>

This tool only accepts a single query at time, each query requires its own separate input.

Input gene lists must contain a **minimum of 15** and a **maximum of 200 genes** matching DREIMdb drug profile genes, otherwise the tool will indicate an error and the analysis will not be performed. Genes are required to be in **gene symbol** format.

#### 4.2 Drug prioritization example<a name="example1"></a>

In this example the input is a csv file and the tool automatically recognizes it as a signature. For demonstrarion purpuses the cells are labeled as "Cell treated with drug" and "Cells untreated".

<img src="assets/help/assets/help/EjemploPrioritization.png" alt="Example prioritization"></img>

After launching the query, a message will indicate the analysis is being perform, in a few minutes the results will appear in the screen. The window can be closed while the analysis is running, the results will be stored, with the name given in the query title, in the Queries history panel.

<img src="assets/help/assets/help/WaitingMsg.png" alt="Waiting"></img>

Once the analysis is performed a page with results will be displayed.

**This page displays three summary plots and a table.**
- **Drug prioritization plot (1)**: plot representing all drugs associated (tau > |75|) with the immune signature. The x-axis represents the tau score, y-axis the -log10(FDR).

 Two areas of interest are depicted in the plot to prioritize drug selection:
 - **Best candidates**: drug associations with |tau| > 90 and a statistically significant FDR (FDR < 0.05).
 - **Good candidates**: drug associations with |tau| > 90 but no statistically significant FDR (FDR > 0.05).

- **Drugs by approval status (2)**: pie-chart representing the status of drugs associated with the immune signature.
- **Drugs by MOA (3)**: pie-chart representing the mode of action of drugs associated with the immune signature.
- **Query parameters (4)**: panel with summarized information of the analysis:
 - **Query up/down genes**: number of user-provided genes in each geneset.
 - **Query up/down genes in DREIMT**: number of user-provided genes from each genesets that match genes in the drug profiles.
- **Drug predictions table (5)**: summarized table with associations (tau > |75|) for the user-provided immune signature, the table can be filtered using the header filters.  Filters applied to this table will update the plots.

<img src="assets/help/assets/help/PriotirizationResults1.png" alt="Prioritization results1"></img>

<img src="assets/help/assets/help/PriotirizationResults1_part2.png" alt="Prioritization results2"></img>

The user-provided signature consisted in 81 upregulated and 78 downregulated genes, of those, 50 upregulated and 45 downregulated genes matched drug profile genes and were used to generate drug associations with the signature. Genes used for the drug association can be downloaded in plain text format by pressing the download query genes button. Result association table can be downloaded as csv format table.












## 5. Signature comparison<a name="comparison"></a>
Signature comparison tool performs geneset-pairwise Fisher's exact test and a Jaccard index analysis to compare a user-provided gene expression signature against DREIMTdb signatures.

#### 5.1 Input<a name="queries-input-comparison"></a>
- **Query title**: (optional) name given to the query. This name will be used to store the results of the query in the Queries history panel.
- **Filters my query**: (optional) filters DREIMTdb signatures  to which theuser-provided signature will be compared. If nothing is specified the comparison will be performed against all DREIMTdb signatures.
- **Perform analysis using only genes in DREIMT**: (optional) specifies the genes used to perform comparison analysis. If marked, only genes in DREIMTdb drug profiles will be used to compare user-provided signature to DREIMTdb signatures. If this option is not marked, all genes will be used to compare user-provided signature to DREIMTdb signatures.
- **Query genes**: user-provided immune signature, can be provided as a list of genes pasted to the text boxes or as a csv/tsv file. For a complete description of input formats see [Inputs file](help#inputs).

<img src="assets/help/assets/help/Panel_Jaccard.png" alt="Panel Jaccard"></img>

Genes can be provided directly to the text box as a **list of genes** or uploaded in a **csv/tsv** format file. Genesets must contain a **minimum of 15** and a **maximum  of 200 genes** , otherwise the tool will indicate an error and the analysis will not be performed. Genes are required to be in **gene symbol** format.

This tool only accepts a single query at time, each query requires its own separate input.

#### 5.2 Signature comparison example I<a name="comparison-example-1"></a>
In this example the user-provided signature will be compared to all DREIMTdb signatures. For demonstrating purposes the the upregulated geneset has been introduced in a single row with genes separated by tabs and the downregulated geneset in a single column format.

<img src="assets/help/assets/help/Jaccard_panel_query1_ppt.png" alt="Panel query1"></img>

After launching the query, a message will indicate the analysis is being perform, in a few minutes the results will appear in the screen. The window can be closed while the analysis is running, the results will be stored, with the name given in the query title, in the Queries history panel.

<img src="assets/help/assets/help/WaitingMsg.png" alt="Waiting message 2"></img>

#### 5.3 Signature comparison example II<a name="comparison-example-1"></a>
In this this example the user-provided is a downregulated signature. The comparison is being perform against DREIMTdb signatures annotated as derived from the comparison of human macrophages stimulated with IL6.

##### Filters applied

 - **Cell type/subtype 1**: Macrophage.
 - **Cell type/subtype 2**: Macrophage.
 - **Organism**: *Homo sapiens*.
 - **Condition**: IL6.

<img src="assets/help/assets/help/Jaccard_panel_query2_ppt.png" alt="Panel query 2"></img>




## 6. Input file<a name="inputs"></a>
Drug prioritization and signature comparison tool take user-provided signatures to prioritize drugs and compase signatures respectively. Signatures can be provide in two formats

#### Input file (csv/tsv)
The input file can either be a **one or two columns** csv/tsv file. Files with more than two columns will not be processed and an error message will be displayed:

- **One column**: for single geneset queries such: upregulated signature, downregulated signature or a geneset without specified direction.
- **Two columns**: for complete signatures (upregulated and downregulated geneset).


The first line must contain column headings that identify the content of each column:
- **GENES_UP**: specifies that the column contains upregulated genes.
- **GENES_DN**: specifies that the column contains downregulated genes.
- **GENES**: specifies that the column contains genes without a specific direction.

Example of the top rows of the four possible csv inputs.

<img src="assets/help/assets/help/FileHeader.png" alt="File header"></img>

#### Input text box
Genes can be introduced to their corresponding text boxes in a single line (whitespaces or tab separated) or as a list (single gene per line).

<img src="assets/help/assets/help/InputBoxes.png" alt="Input boxes"></img>


## 7. Queries history<a name="queries-history"></a>
Drug prioritization and Signature comparison results will be accessible through the Queries history menu. A panel will display the queries performed. Results will be **stored** in DREIMT during **one month**.

<img src="assets/help/assets/help/HistoryPanel.png" alt="History panel"></img>













## 8. Error messages<a name="errors"></a>
The csv/tsv file contains more than two columns.

<img src="assets/help/assets/help/Error3columns.png" alt="Error 1"></img>

The query its out of the required range of a minimum 15 genes and maximum of 200 and/or does not contain (15-200) genes matching drug profile genes.

<img src="assets/help/assets/help/ErrorSize.png" alt="Error 2"></img>

The csv/tsv file header is not specified as expected.

<img src="assets/help/assets/help/ErrorHeader.png" alt="Error 3"></img>










## 9. FAQs<a name="faqs"></a>
#### 9.1 Drugs in DREIMT<a name="f1"></a>
DREIMTdb contains 4,690 different drug profiles of which 3,580 correspond to unique compounds. Duplicated drugs correspond to compounds with different BROAD ID. Because of this, duplicated drugs in DREIMTdb might show different, even opposite tau scores for the same immune signature.
#### 9.2 Tau score change<a name="f2"></a>
Small changes in tau prioritization score can occur after a database update. DREIMTdb prioritizes drug based using all associations, adding new signatures or drugs can produce small changes in tau scores. For the same reason small changes can occur to FDR values.
#### 9.3 Signatures out of the required range<a name="f3"></a>
Drug profiles in DREIMTdb are composed by a ranked list of ~12,000 genes as measured by the LINCS. In order to perform the drug association analysis 15 to 200 immune gene expression signature genes are required to match with drug profile genes. If the user-derived signature is composed by more than 200 we suggest to delimit the query to top 200 genes more significant genes or to those with highest fold change. In case the the signature is composed by less than 15 genes try to include more genes if possible.
#### 9.4 Tau interpretation<a name="f4"></a>
The interpretation of the tau score can, in some cases, be complicated. For signatures, signature up and genesets, a positve tau indicates a boosting effect of the drug over the case cell compared to the reference (or as inhibiting the reference cell compared to the case cell). A negative tau has the opposite interpretation. Contrary a positive tau for a signature down indicate that the drug inhibits the case cell compared to the reference cell.
For this reason, DREIMT includes a summary sentence to ease result interpretation.
#### 9.5 DREIMT terminology<a name="f5"></a>
- **Drug profile**: Gene list rank-ordered according to their differential expression, cell treated with a drug relative to control cells.
- **Signature**: Gene list rank-ordered according to their differential expression, immune cells relative to control. Signatures can be composed either both or one of the next genesets:
 - **Signature up**: geneset composed of the top most signature upregulated genes.
 - **Signature down**: geneset composed of the top most signature downregulated genes.
- **Geneset**: group of genes representing an immune cell without specified control.
